x/marker/keeper/msg_server.go

```diff
@@ -264,12 +264,18 @@ func (k msgServer) Mint(goCtx context.Context, msg *types.MsgMintRequest) (*type
 	}
 
 	admin := sdk.MustAccAddressFromBech32(msg.Administrator)
+	recipient := sdk.MustAccAddressFromBech32(msg.Recipient)
```

Comment:
Only do this if `len(msg.Recipient) > 0` or else it'll panic when no recipient is provided. Similarly, only do the `WithdrawCoins` below if a recipient is provided.